src/detector.js

```diff
@@ -49,11 +49,21 @@ class PhishingDetector {
     return result
   }
 
-  _check (domain) {
+  _check (url) {
+    let domain = url;
+
+    try {
+      domain = new URL(url).host;
```

Comment:
What is this meant to catch? Why would we suddenly have invalid URLs passed in? 🤔

Reply:
Current usage of this passes in a hostname only which will fail and throw a TypeError, so we catch an exception and rebuild with a assumed https protocol

Reply:
But if we are gonna make changes upstream to pass in the full URL, will this still be needed?

Reply:
As per our call, I will rework this section of the code and ping you later today

Reply:
https://github.com/MetaMask/eth-phishing-detect/pull/58432/commits/7e999b46bb0b811ca4d026b2acf359f0d3bc958d made changes as we discussed on the call